### PR TITLE
REF - Switch to using new `CRM_Core_Component::isEnabled()`

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Cleans up code which checks if a component is enabled, using the new utility function created by @eileenmcnaughton in #22567

Technical Details
----------------------------------------
Switches more verbose array searches to the new utility function.

### CODE REVIEW

File: CRM/Contribute/Page/ContributionPage.php [hunk 1]
```diff
@@ -299,15 +299,11 @@ public function run() {
...
-      $config = CRM_Core_Config::singleton();
-
       // assign vars to templates
       $this->assign('id', $id);
       $this->assign('title', CRM_Core_DAO::getFieldValue('CRM_Contribute_DAO_ContributionPage', $id, 'title'));
       $this->assign('is_active', CRM_Core_DAO::getFieldValue('CRM_Contribute_DAO_ContributionPage', $id, 'is_active'));
-      if (in_array('CiviMember', $config->enableComponents)) {
-        $this->assign('CiviMember', TRUE);
-      }
+      $this->assign('CiviMember', CRM_Core_Component::isEnabled('CiviMember'));
```
Comment (colemanw) [inline CRM/Contribute/Page/ContributionPage.php:306]: Unconditional assign to align with our efforts to reduce undefined template vars.

File: CRM/Core/Block.php [hunk 2]
```diff
@@ -349,14 +347,12 @@ private static function setTemplateShortcutValues() {
...
+      if ($components) {
         // check if we can process credit card contribs
         $newCredit = CRM_Core_Config::isEnabledBackOfficeCreditCardPayments();
 
-        foreach ($components as $componentName => $obj) {
-          if (in_array($componentName, $config->enableComponents)) {
-            $obj->creatNewShortcut($shortCuts, $newCredit);
-          }
+        foreach ($components as $obj) {
+          $obj->creatNewShortcut($shortCuts, $newCredit);
```
Comment (colemanw) [inline CRM/Core/Block.php:355]: I had to read this code a couple times to make sense of it. Basically it was fetching a list of enabled components, then for each one, checking to see if it's enabled before taking an action. That's obviously redundant, so I removed the extra check.

File: CRM/Event/Form/ManageEvent.php [hunk 3]
```diff
@@ -88,10 +88,7 @@ public function getDefaultContext() {
    * Set variables up before form is built.
    */
   public function preProcess() {
-    $config = CRM_Core_Config::singleton();
-    if (in_array('CiviEvent', $config->enableComponents)) {
-      $this->assign('CiviEvent', TRUE);
-    }
+    $this->assign('CiviEvent', CRM_Core_Component::isEnabled('CiviEvent'));
```
Comment (colemanw) [inline CRM/Event/Form/ManageEvent.php:91]: Unconditional assign to align with our efforts to reduce undefined template vars.

File: CRM/Activity/BAO/Activity.php [hunk 4]
```diff
@@ -2067,8 +2067,7 @@ public static function getProfileFields() {
       'is_current_revision',
       'activity_is_deleted',
     ];
-    $config = CRM_Core_Config::singleton();
-    if (!in_array('CiviCampaign', $config->enableComponents)) {
+    if (!CRM_Core_Component::isEnabled('CiviCampaign')) {
```
Comment (eileenmcnaughton) [inline CRM/Activity/BAO/Activity.php:2070]: ok

File: CRM/Activity/Task.php [hunk 5]
```diff
@@ -90,8 +90,7 @@ public static function tasks() {
         ],
       ];
 
-      $config = CRM_Core_Config::singleton();
-      if (in_array('CiviCase', $config->enableComponents)) {
+      if (CRM_Core_Component::isEnabled('CiviCase')) {
```
Comment (eileenmcnaughton) [inline CRM/Activity/Task.php:93]: ok

File: CRM/Admin/Form/Options.php [hunk 6]
```diff
@@ -269,8 +269,7 @@ public function buildQuickForm() {
 
     // If CiviCase enabled AND "Add" mode OR "edit" mode for non-reserved activities, only allow user to pick Core or CiviCase component.
     // FIXME: Each component should define whether adding new activity types is allowed.
-    $config = CRM_Core_Config::singleton();
-    if ($this->_gName == 'activity_type' && in_array("CiviCase", $config->enableComponents) &&
+    if ($this->_gName == 'activity_type' && CRM_Core_Component::isEnabled("CiviCase") &&
```
Comment (eileenmcnaughton) [inline CRM/Admin/Form/Options.php:272]: ok

File: ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText/Participant.php [hunk 7]
```diff
@@ -29,9 +29,8 @@ public function __construct() {
    * @return bool
    */
   public function isActive() {
-    $config = CRM_Core_Config::singleton();
-    return in_array('CiviEvent', $config->enableComponents) &&
-    CRM_Core_Permission::check('view event participants');
+    return CRM_Core_Component::isEnabled('CiviEvent') &&
+      CRM_Core_Permission::check('view event participants');
```
Comment (eileenmcnaughton) [inline ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText/Participant.php:33]: ok

File: ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText/Membership.php [hunk 8]
```diff
@@ -29,9 +29,8 @@ public function __construct() {
    * @return bool
    */
   public function isActive() {
-    $config = CRM_Core_Config::singleton();
-    return in_array('CiviMember', $config->enableComponents) &&
-    CRM_Core_Permission::check('access CiviMember');
+    return CRM_Core_Component::isEnabled('CiviMember') &&
+      CRM_Core_Permission::check('access CiviMember');
```
Comment (eileenmcnaughton) [inline ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText/Membership.php:33]: ok

File: ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText/Contribution.php [hunk 9]
```diff
@@ -29,9 +29,8 @@ public function __construct() {
    * @return bool
    */
   public function isActive() {
-    $config = CRM_Core_Config::singleton();
-    return in_array('CiviContribute', $config->enableComponents) &&
-    CRM_Core_Permission::check('access CiviContribute');
+    return CRM_Core_Component::isEnabled('CiviContribute') &&
+      CRM_Core_Permission::check('access CiviContribute');
```
Comment (eileenmcnaughton) [inline ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText/Contribution.php:33]: ok

File: ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText/Case.php [hunk 10]
```diff
@@ -29,8 +29,7 @@ public function __construct() {
    * @return bool
    */
   public function isActive() {
-    $config = CRM_Core_Config::singleton();
-    return in_array('CiviCase', $config->enableComponents);
+    return CRM_Core_Component::isEnabled('CiviCase');
```
Comment (eileenmcnaughton) [inline ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText/Case.php:32]: ok

File: Civi/Api4/Service/Spec/SpecGatherer.php [hunk 11]
```diff
@@ -92,9 +92,7 @@ private function addDAOFields($entity, $action, RequestSpec $spec) {
       if (array_key_exists('contactType', $DAOField) && $spec->getValue('contact_type') && $DAOField['contactType'] != $spec->getValue('contact_type')) {
         continue;
       }
-      if (!empty($DAOField['component']) &&
-        !in_array($DAOField['component'], \Civi::settings()->get('enable_components'), TRUE)
-      ) {
+      if (!empty($DAOField['component']) && !\CRM_Core_Component::isEnabled($DAOField['component'])) {
```
Comment (eileenmcnaughton) [inline Civi/Api4/Service/Spec/SpecGatherer.php:95]: ok

File: Civi/Api4/RelationshipCache.php [hunk 12]
```diff
@@ -53,7 +53,7 @@ public static function getInfo() {
         'description' => ts('One or more related contacts'),
       ],
     ];
-    if (in_array('CiviCase', \Civi::settings()->get('enable_components'), TRUE)) {
+    if (\CRM_Core_Component::isEnabled('CiviCase')) {
```
Comment (eileenmcnaughton) [inline Civi/Api4/RelationshipCache.php:56]: ok

File: CRM/Utils/Check/Component/Env.php [hunk 13]
```diff
@@ -251,9 +251,7 @@ public function checkDefaultMailbox($force = FALSE) {
       return $messages;
     }
 
-    $config = CRM_Core_Config::singleton();
-
-    if (in_array('CiviMail', $config->enableComponents) &&
+    if (CRM_Core_Component::isEnabled('CiviMail') &&
```
Comment (eileenmcnaughton) [inline CRM/Utils/Check/Component/Env.php:254]: ok

File: CRM/Upgrade/Incremental/php/FiveTwenty.php [hunk 14]
```diff
@@ -39,8 +39,7 @@ public function setPreUpgradeMessage(&$preUpgradeMessage, $rev, $currentVer = NU
         ]);
       }
 
-      $config = CRM_Core_Config::singleton();
-      if (in_array('CiviCase', $config->enableComponents)) {
+      if (CRM_Core_Component::isEnabled('CiviCase')) {
```
Comment (eileenmcnaughton) [inline CRM/Upgrade/Incremental/php/FiveTwenty.php:42]: ok

File: CRM/Upgrade/Incremental/php/FiveTwenty.php [hunk 15]
```diff
@@ -62,8 +61,7 @@ public function upgrade_5_20_alpha1($rev) {
       "tinyint(4) DEFAULT '0' COMMENT 'Shows this is a template for recurring contributions.'", FALSE, '5.20.alpha1');
     $this->addTask('Add order_reference field to civicrm_financial_trxn', 'addColumn', 'civicrm_financial_trxn', 'order_reference',
       "varchar(255) COMMENT 'Payment Processor external order reference'", FALSE, '5.20.alpha1');
-    $config = CRM_Core_Config::singleton();
-    if (in_array('CiviCase', $config->enableComponents)) {
+    if (CRM_Core_Component::isEnabled('CiviCase')) {
```
Comment (eileenmcnaughton) [inline CRM/Upgrade/Incremental/php/FiveTwenty.php:64]: ok

File: CRM/Upgrade/Incremental/php/FiveThirteen.php [hunk 16]
```diff
@@ -23,9 +23,7 @@ class CRM_Upgrade_Incremental_php_FiveThirteen extends CRM_Upgrade_Incremental_B
    *   an intermediate version; note that setPostUpgradeMessage is called repeatedly with different $revs.
    */
   public function setPostUpgradeMessage(&$postUpgradeMessage, $rev) {
-    $config = CRM_Core_Config::singleton();
-    $campaignEnabled = in_array("CiviCampaign", $config->enableComponents);
-    if ($rev == '5.13.alpha1' && $campaignEnabled) {
+    if ($rev == '5.13.alpha1' && CRM_Core_Component::isEnabled('CiviCampaign')) {
```
Comment (eileenmcnaughton) [inline CRM/Upgrade/Incremental/php/FiveThirteen.php:26]: ok

File: CRM/Campaign/BAO/Campaign.php [hunk 17]
```diff
@@ -294,7 +294,7 @@ public static function getPermissionedCampaigns(
    * @return bool
    */
   public static function isCampaignEnable(): bool {
-    return in_array('CiviCampaign', CRM_Core_Config::singleton()->enableComponents, TRUE);
+    return CRM_Core_Component::isEnabled('CiviCampaign');
```
Comment (eileenmcnaughton) [inline CRM/Campaign/BAO/Campaign.php:297]: ok

File: CRM/Case/BAO/Case.php [hunk 18]
```diff
@@ -34,8 +34,7 @@ class CRM_Case_BAO_Case extends CRM_Case_DAO_Case {
    * @return bool
    */
   public static function enabled() {
-    $config = CRM_Core_Config::singleton();
-    return in_array('CiviCase', $config->enableComponents);
+    return CRM_Core_Component::isEnabled('CiviCase');
```
Comment (eileenmcnaughton) [inline CRM/Case/BAO/Case.php:37]: ok

File: CRM/Event/Form/ManageEvent.php (hunk 3 above)
Comment (eileenmcnaughton) [reply CRM/Event/Form/ManageEvent.php:91]: yay

**Comment by eileenmcnaughton** [comment]
I finally managed to eyeball all of these changes & they all look OK. Giving merge-ready so this can be merged as soon as the rc is cut (might as well give it that extra month since with so many files changed there is greater risk of  some issue)

**Pull request merged**